dotnet/src/Experimental/Process.LocalRuntime/LocalKernelProcessFactory.cs

```diff
@@ -18,12 +18,9 @@ public static class LocalKernelProcessFactory
     /// <returns>An instance of <see cref="KernelProcess"/> that can be used to interrogate or stop the running process.</returns>
     public static async Task<LocalKernelProcessContext> StartAsync(this KernelProcess process, Kernel kernel, KernelProcessEvent initialEvent)
     {
-        Verify.NotNull(process);
-        Verify.NotNullOrWhiteSpace(process.State?.Name);
-        Verify.NotNull(kernel);
-        Verify.NotNull(initialEvent);
+        Verify.NotNull(initialEvent, nameof(initialEvent));
```

Comment:
in general wondering if for initialEvent there should also be a check for the Id not not be empty

Reply:
This is a good point.  I've updated `KernelProcessEvent.Id` so its no longer nullable, but it might be whitespace or empty string.  I suppose my thought is that this condition is relevant for any event that is emitted, so perhaps the check might be more useful as part of the processing loop.  I've updated `LocalProcess.RunOnceAsync` to check for this (as opposed to the factory) and also `LocalStep.EmitEventAsync` and `KernelProcessStepContext.EmitEventAsync`.